# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaManager.cpp
# Commit: 9a08a35f8cba
# Full Hash: 9a08a35f8cba1b70e3dd912084ce32e12e20994f
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-12 21:43:58
# Description:
#   Bug 1565317 - Handle missing mMediaSource in ReduceConstraint. r=jib
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37778
# ==============================================================================

diff -r c9fe4617e3a4 -r 9a08a35f8cba dom/media/MediaManager.cpp
--- a/dom/media/MediaManager.cpp	Thu Jul 11 18:41:43 2019 +0000
+++ b/dom/media/MediaManager.cpp	Fri Jul 12 13:04:14 2019 +0000
@@ -2289,10 +2289,17 @@
   }
 
   // Keep mediaSource, ignore all other constraints.
-  auto& c = aConstraint.GetAsMediaTrackConstraints();
-  MOZ_DIAGNOSTIC_ASSERT(c.mMediaSource.WasPassed());
-  nsString mediaSource = c.mMediaSource.Value();
-  aConstraint.SetAsMediaTrackConstraints().mMediaSource.Construct(mediaSource);
+  Maybe<nsString> mediaSource;
+  if (aConstraint.GetAsMediaTrackConstraints().mMediaSource.WasPassed()) {
+    mediaSource =
+        Some(aConstraint.GetAsMediaTrackConstraints().mMediaSource.Value());
+  }
+  if (mediaSource) {
+    aConstraint.SetAsMediaTrackConstraints().mMediaSource.Value() =
+        *mediaSource;
+  } else {
+    aConstraint.SetAsMediaTrackConstraints();
+  }
 }
 
 /**
